clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -483,12 +484,19 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
                         parsed.add(parseRecord(tp, logEntry));
                         bytes += logEntry.size();
                     }
+
                     if (!parsed.isEmpty()) {
                         ConsumerRecord<K, V> record = parsed.get(parsed.size() - 1);
                         this.subscriptions.fetched(tp, record.offset() + 1);
                         this.records.add(new PartitionRecords<K, V>(fetchOffset, tp, parsed));
                         this.sensors.recordsFetchLag.record(partition.highWatermark - record.offset());
+                    } else if (buffer.capacity() >= this.fetchSize) {
+                        // we did not read a single message from a max fetchable buffer
+                        // because that message's size is larger than fetch size, in this case
+                        // log an error
+                        log.error("There is a single message whose size is larger than fetch size {} and hence cannot be ever returned", this.fetchSize);
```

Comment:
Yes we will, the design is to let users noticing of the iterative error message to turn off the consumer. I left a comment in the ticket asking if we should gracefully close it upon this case.